### ISSUE TITLE
loss-sensitive training?

### ISSUE THREAD

**Issue by danyaljj** [IllinoisCogComp/illinois-sl#2]
Is there a way to do loss-sensitive training in SL (say with respect to F1, or any other loss that user can define).

**Comment by KaiWeiChang** [comment]
Yes. In the AbstractInferenceSolver.java, you can define your own loss and the corresponding inference method.

**Comment by danyaljj** [comment]
But inside `AbstractInferenceSolver.java` isn't the loss defined [per instance](https://github.com/IllinoisCogComp/illinois-sl/blob/master/src/main/java/edu/illinois/cs/cogcomp/sl/core/AbstractInferenceSolver.java#L104)? If so, that limits the user in defining loss functions, like F1.

**Comment by KaiWeiChang** [comment]
Oh, I see your point. Technically, you can define a huge instance that is a tuple contains all instances (e.g.,http://www.cs.cornell.edu/people/tj/publications/joachims_05a.pdf) and implement the corresponding loss-augmented inference. 

Then, you solve the structure problem with only one instance.
